CMakeLists.txt

```diff
@@ -587,6 +588,111 @@ if (BUILD_TESTING)
             set_property(TEST ${test_target} PROPERTY TIMEOUT 7200)
         endforeach()
     endif()
+
+    if(S2N_FUZZ_TEST)
+        set(SCRIPT_PATH "${CMAKE_CURRENT_SOURCE_DIR}/tests/fuzz/runFuzzTest.sh")
+        file(GLOB FUZZ_TEST_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/tests/fuzz/*.c")
+
+        file(GLOB TESTLIB_SRC "tests/testlib/*.c")
+        file(GLOB TESTLIB_HEADERS "tests/testlib/*.h" "tests/s2n_test.h")
+
+        add_library(fuzztest SHARED ${TESTLIB_HEADERS} ${TESTLIB_SRC})
+        target_include_directories(fuzztest PUBLIC tests)
+        target_link_libraries(fuzztest PUBLIC ${PROJECT_NAME})
+
+        # Set default values for fuzzing if not defined
+        if(NOT DEFINED LIBFUZZER_LIB)
+            set(LIBFUZZER_LIB "/usr/local/libfuzzer/lib/libFuzzer.a")
+        endif()
+        
+        if(NOT DEFINED FUZZ_TIMEOUT_SEC)
+            set(FUZZ_TIMEOUT_SEC 120)
+        endif()
+
+        if(NOT DEFINED CORPUS_UPLOAD_LOC)
+            set(CORPUS_UPLOAD_LOC "none")
+        endif()
+
+        if(NOT DEFINED ARTIFACT_UPLOAD_LOC)
+            set(ARTIFACT_UPLOAD_LOC "none")
+        endif()
+
+        if(NOT DEFINED FUZZ_TESTS)
+            set(FUZZ_TESTS "${TESTS}")
+        endif()
+
+        # Build LD_PRELOAD shared libraries
+        include_directories(
```

Comment:
I think [target_include_directories](https://cmake.org/cmake/help/latest/command/target_include_directories.html#command:target_include_directories) is preferred to `include_directories`, because `include_directories` toggles global things?

If there is a specific reason that we can't use `target_include_directories`, can we add a comment here explaining it?